packages/cli/src/lib/implementation/validate-filter-options.utils.ts

```diff
@@ -76,6 +146,30 @@ export function validateFinalState(
       `Nothing to report. No plugins or categories are available after filtering. Available plugins: ${availablePlugins}. Available categories: ${availableCategories}.`,
     );
   }
+  if (filteredPlugins.every(pluginHasZeroWeightRefs)) {
+    throw new OptionValidationError(
+      `All groups in the filtered plugins have refs with zero weight. Please adjust your filters or weights.`,
+    );
+  }
+}
+
+export function pluginHasZeroWeightRefs(
+  plugin: Pick<PluginConfig, 'groups' | 'audits'>,
+): boolean {
+  if (!plugin.groups || plugin.groups.length === 0) {
+    return false;
+  }
+  const weightMap = new Map<string, number>();
+  plugin.groups.forEach(group => {
+    group.refs.forEach(ref => {
+      weightMap.set(ref.slug, (weightMap.get(ref.slug) ?? 0) + ref.weight);
+    });
+  });
+  const totalWeight = plugin.audits.reduce(
+    (sum, audit) => sum + (weightMap.get(audit.slug) ?? 0),
+    0,
+  );
+  return totalWeight === 0;
```

Comment:
This implementation doesn't take into account that an audit can be referenced by multiple groups within the plugin. I would simply sum up the weights for each group, and then check if some group has a total weight of 0.